include/oneapi/dpl/internal/exclusive_scan_by_segment_impl.h

```diff
@@ -91,18 +92,50 @@ exclusive_scan_by_segment_impl(Policy&& policy, InputIterator1 first1, InputIter
 }
 
 #if _ONEDPL_BACKEND_SYCL
+
 template <typename Policy, typename InputIterator1, typename InputIterator2, typename OutputIterator, typename T,
           typename BinaryPredicate, typename Operator>
 oneapi::dpl::__internal::__enable_if_hetero_execution_policy<typename ::std::decay<Policy>::type, OutputIterator>
-exclusive_scan_by_segment_impl(Policy&& policy, InputIterator1 first1, InputIterator1 last1, InputIterator2 first2,
-                               OutputIterator result, T init, BinaryPredicate binary_pred, Operator binary_op)
+exclusive_scan_by_segment_impl_helper(Policy&& policy, InputIterator1 first1, InputIterator1 last1,
+                                      InputIterator2 first2, OutputIterator result, T init, BinaryPredicate binary_pred,
+                                      Operator binary_op, ::std::true_type /* has_known_identity*/)
 {
-    const auto n = ::std::distance(first1, last1);
+    const auto n = last1 - first1;
 
     // Check for empty element ranges
     if (n <= 0)
         return result;
 
+    namespace __bknd = oneapi::dpl::__par_backend_hetero;
```

Comment:
I think both exclusive and inclusive helpers can be combined in one function with extra parameter that indicates whether the scan is inclusive.

I believe there is no other difference between the code snippets other than exclusive vs inclusive. Please confirm

Reply:
I can confirm this is the case. We still need unique implementations for the no identity case, so I will add a combined function in `scan_by_segment_impl.h` for when a known identity is present and takes inclusive / exclusive as a parameter